src/backend/EdFi.DataManagementService.Backend.Postgresql/Operation/SqlAction.cs

```diff
@@ -648,6 +648,132 @@ TraceId traceId
         return result;
     }
 
+    /// <summary>
+    /// Update the edfidoc of related resources when a child identity is updated
+    /// </summary>
+    public async Task<List<CascadingUpdateResult>> CascadeUpdates(
+        string resourceName,
+        long documentId,
+        short documentPartitionKey,
+        DocumentInfo documentInfo,
+        NpgsqlConnection connection,
+        NpgsqlTransaction transaction,
+        TraceId traceId
+    )
+    {
+        var result = await GetPostgresExceptionRetryPipeline()
+            .ExecuteAsync(async cancellationToken =>
+            {
+                try
+                {
+                    foreach (var id in documentInfo.DocumentIdentity.DocumentIdentityElements)
+                    {
+                        _logger.LogInformation(id.IdentityValue);
+                    }
+
+                    var updateStatement = BuildUpdateStatement(
+                            documentInfo.DocumentIdentity.DocumentIdentityElements.Length - 1
+                        )
+                        .TrimEnd(',');
+
+                    await using var command = new NpgsqlCommand(
+                        @$"update dms.document
+                            set
+                              lastmodifiedat = NOW()
+                              , edfidoc =
+                                {updateStatement}
+                            from (
+                              select parentdocumentid, parentdocumentpartitionkey
+                              from dms.reference
+                              where referenceddocumentid = $1
+                              and referenceddocumentpartitionkey = $2
+                            ) as sub
+                            where document.id = sub.parentdocumentid
+                            and document.documentpartitionkey = sub.parentdocumentpartitionkey
+                            returning document.id as ModifiedDocumentId,
+                                document.documentpartitionkey as ModifiedDocumentPartitionKey,
+                                document.resourcename as ModifiedResourceName;",
+                        connection,
+                        transaction
+                    )
+                    {
+                        Parameters =
+                        {
+                            new() { Value = documentId },
+                            new() { Value = documentPartitionKey },
+                        },
+                    };
+
+                    await command.PrepareAsync(cancellationToken);
+                    await using NpgsqlDataReader reader = await command.ExecuteReaderAsync(cancellationToken);
+
+                    var results = new List<CascadingUpdateResult>();
```

Comment:
If you don't use `var` you can use modern C# like `List<CascadingUpdateResult> results = [];`